extension-manifest-v3/src/background/stats.js

```diff
@@ -265,15 +265,20 @@ if (__PLATFORM__ === 'safari') {
         // with the correct tabId (sometimes it is correct, sometimes it is 0).
         // Thus, let the content_script fire it.
         case 'onCommitted': {
-          const { domain, hostname } = parse(sender.url);
-          setupTabStats(sender.tab.id, domain || hostname);
+          const request = Request.fromRequestDetails({ url: sender.url });
+          setupTabStats(
+            sender.tab.id,
+            request.isHttp || request.isHttps
+              ? request.domain || request.hostname
+              : undefined,
+          );
           break;
         }
         case 'updateTabStats':
           updateTabStats(
             sender.tab.id,
             msg.urls.map((url) =>
-              Request.fromRawDetails({ url, sourceUrl: sender.url }),
+              Request.fromRequestDetails({ url, originUrl: sender.url }),
```

Comment:
shouldn't `originUrl` point to `sender.tab.url` instead? as `sender.url` is likely the iframe url, but I'm not sure if that's true

Reply:
It's just difference between `fromRawDetails` and `fromRequestDetails` - as the second does not have `sourceUrl` but `originUrl` or `documentUrl` (in Firefox). I recommend to use one method everywhere, so we are consistent about generating request meta data, so I have changed it here to `fromRequestDetails`.